py-scripts/lf_ftp.py

```diff
@@ -172,11 +224,98 @@ def __init__(self, lfclient_host="localhost", lfclient_port=8080, sta_prefix="st
         self.channel_list = []
         self.mode_list = []
         self.cx_list = []
+        self.eap_method = eap_method
+        self.eap_identity = eap_identity
+        self.ieee80211 = ieee80211
+        self.ieee80211u = ieee80211u
+        self.ieee80211w = ieee80211w
+        self.enable_pkc = enable_pkc
+        self.bss_transition = bss_transition
+        self.power_save = power_save
+        self.disable_ofdma = disable_ofdma
+        self.roam_ft_ds = roam_ft_ds
+        self.key_management = key_management
+        self.pairwise = pairwise
+        self.private_key = private_key
+        self.ca_cert = ca_cert
+        self.client_cert = client_cert
+        self.pk_passwd = pk_passwd
+        self.pac_file = pac_file
+        self.expected_passfail_val = expected_passfail_val
+        self.csv_name = csv_name
+        self.wait_time = wait_time
+        self.config = config
+        self.group_device_map = {}
+        self.pass_fail_list = []
+        self.test_input_list = []
         self.api_url = 'http://{}:{}'.format(self.host, self.port)
 
         logger.info("Test is Initialized")
 
     def query_realclients(self):
+        config_devices = {}
+        obj = DeviceConfig.DeviceConfig(lanforge_ip=self.host, file_name=self.file_name, wait_time=self.wait_time)
+        upstream = self.change_port_to_ip(self.upstream)
+        config_dict = {
+            'ssid': self.ssid,
+            'passwd': self.password,
+            'enc': self.security,
+            'eap_method': self.eap_method,
+            'eap_identity': self.eap_identity,
+            'ieee80211': self.ieee80211,
+            'ieee80211u': self.ieee80211u,
+            'ieee80211w': self.ieee80211w,
+            'enable_pkc': self.enable_pkc,
+            'bss_transition': self.bss_transition,
+            'power_save': self.power_save,
+            'disable_ofdma': self.disable_ofdma,
+            'roam_ft_ds': self.roam_ft_ds,
+            'key_management': self.key_management,
+            'pairwise': self.pairwise,
+            'private_key': self.private_key,
+            'ca_cert': self.ca_cert,
+            'client_cert': self.client_cert,
+            'pk_passwd': self.pk_passwd,
+            'pac_file': self.pac_file,
+            'server_ip': upstream,
+        }
+        # Case 1: Group name, file name, and profile name are provided, but device list is empty
+        if (self.group_name is not None and self.file_name is not None and self.device_list == [] and self.profile_name is not None):
```

Comment:
Checks for something not being None can be done simply as `if self.group_name:`, for example, rather than the more verbose `if self.group_name is not None:`. More opinionated linters will warn on this.

Additionally, parentheses are not required for Python conditionals.

Please adjust conditional to use simpler not None checks and remove parentheses.